frappe/public/js/frappe/form/linked_with.js

```diff
@@ -120,9 +119,19 @@ frappe.ui.form.LinkedWith = Class.extend({
 			},
 			callback: function(r) {
 				var parent = me.dialog.fields_dict.list.$wrapper.empty();
-
-				if(keys(r.message || {}).length) {
-					$.each(keys(r.message).sort(), function(i, doctype) {
+				var result = {};
+				
+				if (me.link_for) {
```

Comment:
Also filter the query, why make the server get every link when you want only a specific one